lib/presentation/view/pin_code/pin_code_widget.dart

```diff
@@ -0,0 +1,252 @@
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:linshare_flutter_app/presentation/util/helper/string_helper.dart';
+import 'package:linshare_flutter_app/presentation/view/button/button_action_builder.dart';
+import 'package:pin_code_fields/pin_code_fields.dart';
+
+typedef ButtonCallback = void Function(String enteredPin);
+
+class PinCodeWidget extends StatefulWidget {
+  /// Vertical space between widget
+  final double distanceSpaceVertical;
+
+  /// Title properties
+  final String title;
+  final TextStyle titleStyle;
+
+  /// Sub-title properties
+  final String subTitle;
+  final TextStyle subTitleStyle;
+
+  /// Pin fields background color
+  final Color pinCodeFieldBackgroundColor;
+
+  /// Pin fields horizontal padding
+  final double pinCodeFieldPaddingHorizontal;
+
+  /// Total of pin field. Default is 6
+  final int pinCount;
+
+  /// Pin field settings
+  final double pinSizeWidth;
+  final double pinSizeHeight;
+  final double pinBorderRadius;
+  final double pinContentPadding;
+  final double pinMarginHorizontalItem;
+  final TextStyle pinTextStyle;
+  final Color pinBackgroundColor;
+
+  /// Error message settings
+  final String errorText;
+  final TextStyle errorStyle;
+
+  /// Button action settings
+  final String buttonText;
+  final double buttonWidth;
+  final double buttonHeight;
+  final double buttonBorderRadius;
+  final TextStyle buttonStyle;
+  final Color buttonBackgroundColor;
+  final ButtonCallback onButtonClick;
```

Comment:
It is hard to me to use that. But honestly I did not find a better way for that?